### PR TITLE
fix(client) Various fixes to run generator and starter scripts on windows

### PR DESCRIPTION
**Pull request by samwillis**
Included fixes:
- Prefix zod type generator provider with node for exe on windows so that its run with node
- Conditionally escape quotes in docker command on windows, this has to be conditional as it breaks Posix
- Execute the pg-migrations command with npx not yarn

The fixes to the starter are also applied to the web-wa-sqlite example.

---

### To test:

```sh
git clone git@github.com:electric-sql/electric.git && cd electric
git checkout samwillis/windows-fix
pnpm install
cd clients/typescript
pnpm build && pnpm pack
cd ../../examples/web-wa-sqlite/
npm uninstall electric-sql
npm uninstall @prisma/client @electric-sql/prisma-generator prisma  # These are no longer required but we haven't updated the example yet 
npm install ../../clients/typescript/electric-sql-0.8.1.tgz
npm backend:up
npm db:migrate
npm client:generate
npm start
```

If for some reason the client won't build on windows, this is a ready made build from this branch: 
[electric-sql-0.8.1.tgz](https://github.com/electric-sql/electric/files/13635086/electric-sql-0.8.1.tgz) - just try it with the web-wa-sqlite *in this branch*

### CODE REVIEW

File: clients/typescript/src/cli/migrations/migrate.ts [hunk 1]
```diff
@@ -235,7 +235,7 @@ async function createPrismaSchema(
   const output = path.resolve(out)
   const schema = dedent`
     generator electric {
-      provider      = "${escapePathForString(provider)}"
+      provider      = "node ${escapePathForString(provider)}"
```
Comment (balegas) [inline clients/typescript/src/cli/migrations/migrate.ts:238]: This worked

**Review by balegas — CHANGES_REQUESTED** [review]
The last issue I found is in `"start": "yarn electric:check && SERVE=true npm run build"`
Windows shell does not support using env vars that way.  This `$env:SERVE="true";`works, but maybe there is a better way.

Thanks for working on this

File: examples/starter/template/db/migrate.js [hunk 2]
```diff
@@ -4,8 +4,8 @@ const process = require('process')
 
 console.info(`Connecting to proxy at ${PUBLIC_DATABASE_URL}`)
 
-const args = ["run", "-s", "pg-migrations", "apply", "--database",  DATABASE_URL, "--directory", "./db/migrations"]
-const proc = spawn("yarn", args, { cwd: __dirname, stdio: ['inherit', 'pipe', 'inherit']  })
+const args = ["pg-migrations", "apply", "--database",  DATABASE_URL, "--directory", "./db/migrations"]
+const proc = spawn("npx", args, { stdio: ['inherit', 'pipe', 'inherit']  })
```
[the remark this answers is not in the document]
Comment (alco) [reply examples/starter/template/db/migrate.js:8]: Correct me if I'm wrong, but when you pass the `shell` option, the first argument to `spawn()` should be a string that includes all arguments. Basically, a one-liner you would type into your shell by hand.

[the remark this answers is not in the document]
Comment (samwillis) [reply examples/starter/template/db/migrate.js:8]: Looks like Node handles this itself, if there are arguments and shell=true then it constructs the command by joining the array of args with a space:

https://github.com/nodejs/node/blob/8cdb7cae98468efee222727ad9a4096e415f3ef6/lib/child_process.js#L620-L645

**Review by kevin-dp — APPROVED** [review]
LGTM!

**Pull request merged**